pkg/generator/vault/vault.go

```diff
@@ -69,6 +69,12 @@ func (g *Generator) generate(ctx context.Context, c *provider.Connector, jsonSpe
 	if res == nil || res.Spec.Provider == nil {
 		return nil, fmt.Errorf("no Vault provider config in spec")
 	}
+
+	class := res.Spec.Controller
+	if class != "" && class != controllerClass {
+		return nil, fmt.Errorf("skipping Vault Dynamic Secret as it points to a unmanaged controllerClass")
```

Comment:
Hi @rdeepc, I had doubt about this part. I'm wondering if there's a better way to handle it, maybe something similar to what the SecretStore is doing: https://github.com/external-secrets/external-secrets/blob/6a2a050928b016b58ae540646b464eb2759ba939/pkg/controllers/secretstore/common.go#L44-L47 An error might be a bit extreme, just informing the user should be enough?

Reply:
Hi @fad3t, 

That's a good idea, as @moolen suggested, I will change the implementation for a generalized solution.